### PR TITLE
PoC run integrated test tiger UI-SDK

### PR DESCRIPTION
**Pull request by maitrieuphilong**
---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10900/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12987/ : FAILURE in 11m 26s

File: run-integrated-tiger.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
```
Comment (caohhung) [inline run-integrated-tiger.sh:1]: This file should be 

- moved to `common/scripts/ci` . and file name should be s.t like `run-sdk-backend-tiger-integration-tests-with-live-backend.sh`
- Or moved to `libs/sdk-backend-tiger/scripts` and replace `integrated-test-AIO.sh`

File: tools/plugin-toolkit/src/_base/env.ts [hunk 2]
```diff
@@ -42,8 +42,9 @@ export function loadEnv(backend: TargetBackendType): Record<string, string> {
         }
     }
 
-    if (backend === "tiger" && TIGER_API_TOKEN) {
+    if (backend === "tiger" && TIGER_API_TOKEN && TEST_EMPTY_WORKSPACE_ID) {
         dotEnvContent.TIGER_API_TOKEN = TIGER_API_TOKEN;
+        dotEnvContent.TEST_EMPTY_WORKSPACE_ID = TEST_EMPTY_WORKSPACE_ID;
```
Comment (caohhung) [inline tools/plugin-toolkit/src/_base/env.ts:47]: Please introduce a nested `if` condition for `WORKSPACE_ID`? do you know why?

File: run-integrated-tiger.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+sleep 120
+# HOST=$(docker inspect "gooddata-cn-ce-manual" | jq -r '.[0].NetworkSettings.IPAddress')
+
+export DB_HOST=gooddata-cn-ce-manual \
+export DB_PORT=5432 \
+export DB_USER=demouser \
+export DB_NAME=demo \
+export DB_PASSWORDS="{'pg_local': 'demopass'}"
+docker run -it --rm -e DB_HOST -e DB_PORT -e DB_USER -e DB_NAME -e DB_PASSWORDS --entrypoint '' --network aio-net \
+441851189095.dkr.ecr.eu-central-1.amazonaws.com/nas/data-loader:master /bin/bash -c "/data_load/data_load.py -w goodsales -d pg_local --debug  --skip-download --no-schema-versioning -f"
```
Comment (caohhung) [inline run-integrated-tiger.sh:25]: Best practice - To reduce the complexity + level of arg passing:
```
docker run -it --rm \
  -e DB_PORT=5432 \
 .....

File: run-integrated-tiger.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+export HOST=http://localhost:3000
+
+curl $HOST/api/v1/entities/dataSources \
+  -H "Content-Type: application/vnd.gooddata.api+json" \
+  -H "Accept: application/vnd.gooddata.api+json" \
+  -H "Authorization: Bearer YWRtaW46Ym9vdHN0cmFwOmFkbWluMTIz" \
+  -X POST \
+  -d '{
```
Comment (caohhung) [inline run-integrated-tiger.sh:34]: Do you think having this body in file is better?

File: run-integrated-tiger.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+export DB_PASSWORDS="{'pg_local': 'demopass'}"
+docker run -it --rm -e DB_HOST -e DB_PORT -e DB_USER -e DB_NAME -e DB_PASSWORDS --entrypoint '' --network aio-net \
+441851189095.dkr.ecr.eu-central-1.amazonaws.com/nas/data-loader:master /bin/bash -c "/data_load/data_load.py -w goodsales -d pg_local --debug  --skip-download --no-schema-versioning -f"
+
+export HOST=http://localhost:3000
+
+curl $HOST/api/v1/entities/dataSources \
+  -H "Content-Type: application/vnd.gooddata.api+json" \
+  -H "Accept: application/vnd.gooddata.api+json" \
+  -H "Authorization: Bearer YWRtaW46Ym9vdHN0cmFwOmFkbWluMTIz" \
```
Comment (caohhung) [inline run-integrated-tiger.sh:32]: Please set `YWRtaW46Ym9vdHN0cmFwOmFkbWluMTIz` as variable in the top of this script, as it's called at least twice

File: run-integrated-tiger.sh [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  now=$(date '+%Y-%m-%d %H:%M:%S')
+  echo "$now $*" 1>&2
+}
+
+# docker network create aio-net
+
+docker run --name gooddata-cn-ce-manual -e APP_LOGLEVEL=INFO -e LICENSE_AND_PRIVACY_POLICY_ACCEPTED=YES \
+    -e BUNDLE_TYPE=gdc -p 3000:3000 -p 5432:5432 --network aio-net -d gooddata/gooddata-cn-ce:2.2.0
+sleep 120
+# HOST=$(docker inspect "gooddata-cn-ce-manual" | jq -r '.[0].NetworkSettings.IPAddress')
```
Comment (caohhung) [inline run-integrated-tiger.sh:17]: remove comment

File: run-integrated-tiger.sh [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  local now;
+  now=$(date '+%Y-%m-%d %H:%M:%S')
+  echo "$now $*" 1>&2
+}
+
+# docker network create aio-net
+
+docker run --name gooddata-cn-ce-manual -e APP_LOGLEVEL=INFO -e LICENSE_AND_PRIVACY_POLICY_ACCEPTED=YES \
+    -e BUNDLE_TYPE=gdc -p 3000:3000 -p 5432:5432 --network aio-net -d gooddata/gooddata-cn-ce:2.2.0
+sleep 120
```
Comment (caohhung) [inline run-integrated-tiger.sh:16]: Replace hard sleep with polling an AIO endpoint

File: run-integrated-tiger.sh [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+log() {
+  local now;
+  now=$(date '+%Y-%m-%d %H:%M:%S')
+  echo "$now $*" 1>&2
+}
+
+# docker network create aio-net
+
+docker run --name gooddata-cn-ce-manual -e APP_LOGLEVEL=INFO -e LICENSE_AND_PRIVACY_POLICY_ACCEPTED=YES \
```
Comment (caohhung) [inline run-integrated-tiger.sh:14]: Container name should be unique -> Add `${EXECUTOR_NUMBER}` 
And should be a variables ?

File: run-integrated-tiger.sh [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+# run the tests
+npm run integrated-test-AIO
+
+remove previously created workspace
+if [ -n "$WORKSPACE_CREATED" ]; then
+    npm run delete-ref-workspaces
+fi
+docker stop gooddata-cn-ce-manual
+docker rm gooddata-cn-ce-manual
```
Comment (caohhung) [inline run-integrated-tiger.sh:78]: Please set them in `trap` in top of script. i.e
```trap "(echo command_1 ; echo command_2) || true" EXIT

Consider to use `docker rm -f` :  `-f, --force     Force the removal of a running container (uses SIGKILL)`

File: run-integrated-tiger.sh [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+cd ./libs/sdk-backend-tiger
+export TIGER_API_TOKEN=YWRtaW46Ym9vdHN0cmFwOmFkbWluMTIz
+export SDK_BACKEND=TIGER
+export TIGER_DATASOURCES_NAME=pg_staging-goodsales
+
+# check if workspace exists or not
+if [[ -z "${TEST_EMPTY_WORKSPACE_ID}" ]]; then
+  # export TEST_EMPTY_WORKSPACE_ID
+  npm run create-ref-workspaces
```
Comment (caohhung) [inline run-integrated-tiger.sh:59]: Problem? `npm run` can be run in jenkins slave.
To use npm/node in jenkins-slave, a node docker images is needed

File: run-integrated-tiger.sh (hunk 8 above)
Comment (caohhung) [reply run-integrated-tiger.sh:14]: Same for `network`, it should be created with unique name + put into `trap` to be removed later.

`trap`: see comments below

**Review by caohhung — CHANGES_REQUESTED** [review]

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Pull request closed without merging**